cpp/inc/bond/core/bonded.h

```diff
@@ -26,6 +26,22 @@ template <typename Protocols, typename Transform, typename T, typename Reader>
 typename boost::enable_if<need_double_pass<Transform>, bool>::type inline
 ApplyTransform(const Transform& transform, const bonded<T, Reader>& bonded);
 
+template <typename T, typename U>
+typename boost::enable_if<std::is_reference<T>, T>::type
+inline move_data(U& data)
```

Comment:
Please add comments for why this is needed ) Other readers could not understand the issue and the solution )